### PR TITLE
Generate pre-release analysis files

### PR DESCRIPTION
**Pull request by ewafula**
#### What scientific question is your analysis addressing?
bash script to generate pre-lease analysis files 

#### What was your approach?
Bash script to run pre-release analysis modules described in the ticket and upload to the s3 bucket

#### What GitHub issue does your pull request address?
- https://github.com/PediatricOpenTargets/ticket-tracker/issues/402

#### Which areas should receive a particularly close look?
Review the entire bash script code and look through pre-release data files upload on the s3 bucket:
- https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/open-targets/analysis-files-pre-release
- https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/open-targets/analysis-files-pre-release-commit

#### Is there anything that you want to discuss further?
Additional analyses modules were updated that were not running properly
- independent-samples module (minor edit, failing load base histologies files)
- tp53_nf1_score (several script amended)

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
Tables 

#### What is your summary of the results?
- https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/open-targets/analysis-files-pre-release
- https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/open-targets/analysis-files-pre-release-commit

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
@zhangb1 and @yuankunzhu can you review this as a script which can be run in one swoop on EC2 to generate all release data files post initial matrix generation (eg, SNV consensus MAF + fusion, CNV files from each algorithm)? Thanks

**Review by zhangb1 — APPROVED** [review]

File: analyses/tp53_nf1_score/00-tp53-nf1-alterations.R [hunk 1]
```diff
@@ -85,6 +85,9 @@ gencode_cds <- read_tsv(gencodeBed, col_names = FALSE)
 # histology file
 histology <- read_tsv(histologyFile, guess_max = 100000)
 
+# recording of "BS_ETC8R0TD" GMKF NBL sample; needs to be remove once updated in v12
+histology$RNA_library[histology$Kids_First_Biospecimen_ID == "BS_ETC8R0TD"] <- "stranded"
```
Comment (jharenza) [inline analyses/tp53_nf1_score/00-tp53-nf1-alterations.R:89]: I created [this](https://github.com/PediatricOpenTargets/ticket-tracker/issues/408) ticket to track

File: scripts/generate-analysis-files.sh [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+cp ${analyses_dir}/tmb-calculation/results/snv-mutation-tmb-coding.tsv ${release_dir}
+cp ${analyses_dir}/tmb-calculation/results/snv-mutation-tmb-all.tsv ${release_dir}
+
+# Run GSEA
+echo "Run GSEA"
+cd ${analyses_dir}/gene-set-enrichment-analysis
+OPENPBTA_BASE_SUBTYPING=1 bash run-gsea.sh
+
+# Copy over GSEA results for subtyping
+cp ${analyses_dir}/gene-set-enrichment-analysis/results/gsva_scores.tsv ${release_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files.sh]: I think this should go in the commit directory?

File: scripts/generate-analysis-files.sh [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+# Copy over independent specimen lists
+cp ${analyses_dir}/independent-samples/results/independent-specimens.*  ${release_dir}
+
+# Run fusion summary
+echo "Run fusion summary for subtypes"
+cd ${analyses_dir}/fusion-summary
+bash run-new-analysis.sh
+
+# Copy over fusion summary
+cp ${analyses_dir}/fusion-summary/results/* ${release_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files.sh:107]: Can you also add these to the commit directory?

File: scripts/generate-analysis-files.sh [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  RUN_ORIGINAL=1 OPENPBTA_BASE_SUBTYPING=1 bash run-prepare-cn-OpenTarget.sh
+  
+  ## Copy over focal CN
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_wgs_plus_cnvkit_wxs_autosomes.tsv.gz ${release_dir}
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_wgs_plus_cnvkit_wxs_x_and_y.tsv.gz ${release_dir}
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_wgs_plus_cnvkit_wxs.tsv.gz ${release_dir}
+
+  # Move over the consensus with status file
+  cp ${analyses_dir}/focal-cn-file-preparation/results/consensus_seg_with_status.tsv ${release_dir}
+  cp ${analyses_dir}/focal-cn-file-preparation/results/cnvkit_with_status.tsv ${release_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files.sh]: Can you also add these to the commit directory?

File: scripts/generate-analysis-files.sh [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+cd ${analyses_dir}/independent-samples
+OPENPBTA_BASE_SUBTYPING=1 bash run-independent-samples.sh
+
+# Run fusion filtering
+echo "Create fusion filtered list"
+cd ${analyses_dir}/fusion_filtering
+OPENPBTA_BASE_SUBTYPING=1 bash run_fusion_merged.sh
+
+# Copy over fusions lists
+cp ${analyses_dir}/fusion_filtering/results/fusion-putative-oncogenic.tsv ${release_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files.sh:51]: Can you also add these to the commit directory?

**Review by jharenza — CHANGES_REQUESTED** [review]
Thanks for putting this together. I had a few suggestions of files to keep for QC checks and then think this PR should also delete `[generate-analysis-files-repo.sh](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/scripts/generate-analysis-files-repo.sh)` and [generate-analysis-files-toolkit.sh](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/scripts/generate-analysis-files-toolkit.sh) and update [run-for-subtyping.sh](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/scripts/run-for-subtyping.sh) to remove the modules now being run upstream.

File: analyses/tp53_nf1_score/00-tp53-nf1-alterations.R (hunk 1 above)
Comment (ewafula) [reply analyses/tp53_nf1_score/00-tp53-nf1-alterations.R:89]: Thanks!

**Comment by ewafula** [comment]
@jharenza, I have made the changes you suggested and included all the subtyping modules that were in the `scripts/run-for-subtyping.sh` script.  I have detected `scripts/generate-analysis-files-toolkit.sh`,  `scripts/generate-analysis-files-toolkit.sh` and `scripts/run-for-subtyping.sh` scripts.  Results for all subtyping modules are only copied to the s3  bucket `pre-release-commit` folder. 

With this PR commit, I have commented out commands to upload files to the s3 bucket `pre-release` and `pre-release-commit` folders for you to check first and confirm that files are being copied to the correct folder.  Once you approve the PR,  I'll rerun the script to upload subtyping files to the `pre-release-commit`  folder.  

I don't have delete permissions on the s3 bucket. Please delete the current content of the `analysis-files-pre-release`  folder as it contains the `tp53` and `gsea` subtyping files that I had previously uploaded with the initial  PR commit. After PR has been fully approved, I will re-upload the pre-release files with a rerun of the script.

**Comment by jharenza** [comment]
Hi @ewafula can you do a clean rerun of the TP53 module in this branch? I am noticing multiple scores for the same BS_ID. EG:

Screen Shot 2022-09-26 at 2 55 27 PM

**Comment by ewafula** [comment]
> Hi @ewafula can you do a clean rerun of the TP53 module in this branch? I am noticing multiple scores for the same BS_ID. EG:
> 
> Screen Shot 2022-09-26 at 2 55 27 PM

@jharenza,  duplicates BS_ids are a result of the following code snippet: 
https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/65c8be222c98162099fb1446c2dc9b7ceccb9ff1/analyses/tp53_nf1_score/02-qc-rna_expression_score.Rmd#L41-L62

Here the match pattern reads classifier results (`pattern="*classifier_scores.tsv"`), including all of the following:
```
gene-expression-rsem-tpm-collapsed_classifier_scores.tsv
gene-expression-rsem-tpm-collapsed-exome_capture_classifier_scores.tsv
gene-expression-rsem-tpm-collapsed-poly-A_classifier_scores.tsv
gene-expression-rsem-tpm-collapsed-poly-A-stranded_classifier_scores.tsv
gene-expression-rsem-tpm-collapsed-stranded_classifier_scores.tsv
```

I think we need to exclude `gene-expression-rsem-tpm-collapsed_classifier_scores.tsv`,  a remnant result file from older runs that has never been removed and is the cause of duplicate BS_ids.  I'll delete the file and rerun the script.

**Comment by jharenza** [comment]
great, thank you!

**Comment by jharenza** [comment]
>I don't have delete permissions on the s3 bucket. Please delete the current content of the analysis-files-pre-release folder as it contains the tp53 and gsea subtyping files that I had previously uploaded with the initial PR commit. After PR has been fully approved, I will re-upload the pre-release files with a rerun of the script.

done

File: scripts/generate-analysis-files.sh [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+# Copy over GSEA results for subtyping
+cp ${analyses_dir}/gene-set-enrichment-analysis/results/gsva_scores.tsv ${commit_dir}
+
+# Run TP53
+echo "TP53 altered score"
+cd ${analyses_dir}/tp53_nf1_score
+OPENPBTA_BASE_SUBTYPING=1 bash run_classifier.sh
+
+# Copy over TP53 results
+cp ${analyses_dir}/tp53_nf1_score/results/*.tsv ${commit_dir}
```
Comment (jharenza) [inline scripts/generate-analysis-files.sh]: Let's also add the ROC plots

**Review by jharenza — APPROVED** [review]
I will pre-approve now and just double check the TP53 results and commit directory once rerun!

File: scripts/generate-analysis-files.sh (hunk 6 above)
Comment (ewafula) [reply scripts/generate-analysis-files.sh]: > Let's also add the ROC plots

Ok, will include plots.

**Comment by ewafula** [comment]
@jharenza regenerated the results but have `Access Denied` to upload to the s3 bucket. I will reach out to Chris to reset permission again.   Chris has been resetting permissions whenever this happens. 

I already updated the PR with the bash script without results for modules run. There are too many files,  some larger than allowed on Github.

**Comment by ewafula** [comment]
@jharenza, I ran the bash script with the `RUN_LOCAL=1` variable to exclude the large files produced by the `run-gistic` and `focal-cn-file-preparation` modules. Results for all other modules included in the script are uploaded to GitHub with the PR update. I'll wait for Chris to resolve permission issues with my access to the s3 bucket to upload all the files to their respective folders.

**Comment by ewafula** [comment]
@jharenza, Chris to resolve permission issues. Files are now uploaded to the s3 bucket.

**Comment by jharenza** [comment]
@ewafula thank you for working on this - I think it looks good. My only remaining comment is maybe we can only retain the `*definining_lesions` and `*molecular_subtype` files for the subtyping modules, rather than each individual lesion file, but we can update that later as needed. May discuss this also with @sickler-alex and @aadamk - maybe once we have the toolkit developed, we can have these outputs committed as well.

**Pull request merged**